lib/modules/datasource/custom/index.ts

```diff
@@ -25,26 +26,45 @@ export class CustomDatasource extends Datasource {
       return null;
     }
 
-    const { defaultRegistryUrlTemplate, transformTemplates, format } = config;
-    let response: unknown;
-    try {
+    const {
+      isLocalRegistry,
+      defaultRegistryUrlTemplate,
+      transformTemplates,
+      format,
+    } = config;
+
+    let data: unknown;
+    if (isLocalRegistry) {
       switch (format) {
         case 'plain':
-          response = await plainFetch(this.http, defaultRegistryUrlTemplate);
+          data = await plainRead(defaultRegistryUrlTemplate);
           break;
         case 'yaml':
-          response = await yamlFetch(this.http, defaultRegistryUrlTemplate);
+          data = await yamlRead(defaultRegistryUrlTemplate);
           break;
         case 'json':
-          response = (await this.http.getJson(defaultRegistryUrlTemplate)).body;
+          data = await jsonRead(defaultRegistryUrlTemplate);
+          break;
+      }
+    } else {
+      try {
+        switch (format) {
+          case 'plain':
+            data = await plainFetch(this.http, defaultRegistryUrlTemplate);
+            break;
+          case 'yaml':
+            data = await yamlFetch(this.http, defaultRegistryUrlTemplate);
+            break;
+          case 'json':
+            data = await jsonFetch(this.http, defaultRegistryUrlTemplate);
+            break;
+        }
+      } catch (e) {
+        this.handleHttpErrors(e);
+        return null;
       }
```

Comment:
Self-reviewing: maybe the if/else isn't ideal, I could move it inside each format as a parameter to simplify the logic in this controller - but unsure
Also, my first implementation was to keep one big switch and use ternary in cases but it would throw in the catch http block which isn't ideal